File: Original Artifact/src/Task/Task.java
package Task;

public class Task {
	
	private String id;
	private String name;
	private String description;
	
	public Task(String id,String name,String description) {
		if(id == null || id.length()>10) {
			throw new IllegalArgumentException("Invalid id");
		}
		if(name == null || name.length()>20) {
			throw new IllegalArgumentException("Invalid Name");
		}
		if(description == null || description.length()>50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		
		this.id = id;
		this.name = name;
		this.description = description;
	}
	
	public String getId() {
		return id;
	}
	public String getName() {
		return name;
	}
	public void setName(String Name) {
		if(name == null || name.length()>20) {
			throw new IllegalArgumentException("Invalid Name");
		}
		this.name = Name;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String Description) {
		if(name == null || name.length()>50) {
			throw new IllegalArgumentException("Invalid Description");
		}
		this.description = Description;
	}
}

File: Enhanced Artifact/task.py
# -*- coding: utf-8 -*-
"""

File: Task.py
Author: Paul Stephan
Version: 1.0
Date: Sep 22, 2024

Description:
This class represents a task with details such as ID, name, and description.

"""

class Task:
    # Constructor to initialize the task object with id, name, and description
    def __init__(self, task_id, name, description):
        # Validation checks for the input parameters
        if not task_id or len(task_id) > 10:
            raise ValueError("Invalid ID")
        if not name or len(name) > 20:
            raise ValueError("Invalid Name")
        if not description or len(description) > 50:
            raise ValueError("Invalid Description")

        # Assign values to attributes
        self.id = task_id
        self.name = name
        self.description = description

    # Getter method for retrieving the task ID
    def get_id(self):
        return self.id

    # Getter method for retrieving the task name
    def get_name(self):
        return self.name

    # Setter method for updating the task name
    def set_name(self, name):
        if not name or len(name) > 20:
            raise ValueError("Invalid Name")
        self.name = name

    # Getter method for retrieving the task description
    def get_description(self):
        return self.description

    # Setter method for updating the task description
    def set_description(self, description):
        if not description or len(description) > 50:
            raise ValueError("Invalid Description")
        self.description = description